cpp/src/lists/copying/segmented_gather.cu

```diff
@@ -38,35 +40,46 @@ std::unique_ptr<column> segmented_gather(lists_column_view const& value_column,
   CUDF_EXPECTS(value_column.size() == gather_map.size(),
                "Gather map and list column should be same size");
 
-  auto gather_map_sliced_child = gather_map.get_sliced_child(stream);
-  auto const gather_map_size   = gather_map_sliced_child.size();
-  auto gather_index_begin      = gather_map.offsets().begin<size_type>() + 1 + gather_map.offset();
-  auto gather_index_end        = gather_index_begin + gather_map.size();
-  auto value_offsets           = value_column.offsets().begin<size_type>() + value_column.offset();
-  auto map_begin = cudf::detail::indexalator_factory::make_input_iterator(gather_map_sliced_child);
+  auto const gather_map_sliced_child = gather_map.get_sliced_child(stream);
+  auto const gather_map_size         = gather_map_sliced_child.size();
+  auto const gather_index_begin = gather_map.offsets().begin<size_type>() + 1 + gather_map.offset();
```

Comment:
Why do we need to have `+1` here?
```suggestion
  auto const gather_index_begin = gather_map.offsets_begin() + 1;
```

Reply:
> Why do we need to have `+1` here?

@karthikeyann might answer this better than I. (The index math is unchanged in this PR.) I think this might be because the first gather index can be retained unchanged?

I'll rephrase to use `offset_begin()` and `offset_end()`.

Reply:
because of the way `gather_index_begin` is used in `thrust::uppper_bound` in the code.